File: R/get_shuttle_data.R
# set file path
df <- read.csv("your_file.csv", stringsAsFactors = FALSE)

urls <- df$download_link
urls <- urls[!is.na(urls) & urls != ""]

# Output folder?
outdir <- "downloads"
dir.create(outdir, showWarnings = FALSE)

for (url in urls) {

  clean_url <- strsplit(url, "\\?")[[1]][1]
  filename <- basename(clean_url)
  destfile <- file.path(outdir, filename)

  tryCatch({
    download.file(url, destfile = destfile, mode = "wb")
    cat("Downloaded:", filename, "\n")
  }, error = function(e) {
    cat("Failed:", url, "\n")
  })
}

File: python/get_shuttle_data.py
import pandas as pd
import requests
import os
from urllib.parse import urlparse

# set file path
df = pd.read_csv("your_file.csv")

urls = df["download_link"].dropna()

outdir = "downloads"
os.makedirs(outdir, exist_ok=True)

for url in urls:
    try:
        clean_path = urlparse(url).path
        filename = os.path.basename(clean_path)
        filepath = os.path.join(outdir, filename)

        r = requests.get(url, stream=True)
        r.raise_for_status()

        with open(filepath, "wb") as f:
            for chunk in r.iter_content(chunk_size=8192):
                f.write(chunk)

        print("Downloaded:", filename)

    except Exception as e:
        print("Failed:", url, "| error:", e)
